src/java/org/apache/nutch/util/NutchJob.java

```diff
@@ -25,8 +25,9 @@
 /** A {@link Job} for Nutch jobs. */
 public class NutchJob extends Job {
 
-  public NutchJob(Configuration conf) throws IOException {
-    super(conf, "NutchJob");
-  }
+  public static NutchJob getJobInstance(Configuration conf){
```

Comment:
This latest addition will still give error because we do not have an constructor in this class and JVM will add a default constructor which will have a default super call to the parent class. Something of this form : 
***default constructor***
public NutchJob(){
  super()
}

But this won't create any discrepancy because we are not utilizing that default constructor in the whole code base.